### PR TITLE
fix: retry on failure and throw correct errors

### PR DESCRIPTION
**Pull request by digitalsadhu**
## JIRA Issue
[COREWEB-107](https://jira.finn.no/browse/COREWEB-107)

## Status
**READY**

## Description
Adds retry logic to get and set so that this does not need to be housed in the build server.

## Todos
- [x] Tests - caveat, I should have tested retry logic, however its brutal to try to set up tests for this against google cloud at the current moment. It's both painful to try to use nock when there are multiple steps (including auth) etc and the current mock implementation doesn't make it easy either so i'm moving on for now. I think this should be revisited.
- [ ] Documentation - N/A

### CODE REVIEW

File: lib/sink.js [hunk 1]
```diff
@@ -203,11 +223,22 @@ module.exports = class SinkGCS extends EventEmitter {
...
-        await this[getGcsFile](fileName).save(fileContent, {
-            metadata: {
-                contentType,
-            },
-        });
+        const file = this[getGcsFile](fileName);
+        const saveFile = () =>
+            file.save(fileContent, { metadata: { contentType } });
+        try {
+            return await pRetry(saveFile, { retries: 3 });
```
Comment (SimenB) [inline lib/sink.js]: I'm not sure we should retry save, in the same way you should not retry `POST`. @ivarconr thoughts?

**Comment by SimenB** [comment]
>It's both painful to try to use nock when there are multiple steps (including auth) etc

How so? You just set it up once and say `.times(n)`, right?

**Comment by digitalsadhu** [comment]
Its the setup. I find it pretty painful. I can try again with record and see how far I get if you insist but I'm not spending more than say 30 mins on it. I want to end the week on a happy note ;)

**Comment by SimenB** [comment]
Fair enough 🙂 
It's not our retry logic, so no real need to test it here. more interesting for the integration test repo, I suppose

File: lib/sink.js (hunk 1 above)
Comment (digitalsadhu) [reply lib/sink.js]: k. We did see cases of upload failures though so we would back at not fully handling our issues.

Comment (SimenB) [reply lib/sink.js]: we have retry in the client on the other side, though.

having multiple layers of retry doesn't scale.

Comment (digitalsadhu) [reply lib/sink.js]: yeah. true dat

Comment (digitalsadhu) [reply lib/sink.js]: How should we resolve this one? Ask @ivarconr for comment again?

**Comment by digitalsadhu** [comment]
Merging, can revisit if we decide anything is wrong later

**Pull request merged**